### ISSUE TITLE
Generic function arguments not checked

### ISSUE THREAD

**Issue by dphilipson** [microsoft/pyright#266]
In some circumstances, generic type parameters aren't checked for consistency.

**To Reproduce**

```python
T = TypeVar("T")

def for_each(xs: List[T], f: Callable[[T], Any]) -> None:
    for x in xs:
        f(x)

def call_len(x: str) -> None:
    len(x)

for_each([1, 2, 3], call_len)
```
The last line passes typechecking, but fails at runtime with
```
TypeError: object of type 'int' has no len()
```

**Expected behavior**

The last line should fail typechecking.

**VS Code extension or command-line**

VSCode extension, version 1.0.65.

**Comment by erictraut** [comment]
Thanks for the bug report. This was a recent regression in the type checker — one that I obviously didn't catch in the unit tests. I've fixed the problem and added your example as a test so it won't happen again. The fix will be in the next version of pyright.

**Comment by dphilipson** [comment]
Appreciate it! Love your work.

**Comment by erictraut** [comment]
This is now fixed in 1.0.66, which I just published.
